llvm/test/Transforms/LoopVectorize/multicmp.ll

```diff
@@ -0,0 +1,794 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
```

Comment:
Could you pre-commit this test so we can see the difference your patch makes?

Reply:
Thanks, Sam. I just added test description or C representations. Is it ok?

Reply:
The comments are useful but I meant that you could commit the tests to main and rebase this PR on top of that so that we can see the difference this PR makes. Florian made a similar suggestion but with a new PR instead of committing straight to main.

Reply:
Or just take the simplest option of having two commits in this patch. Commit 1 - add tests, Commit 2 - add the code change which shows the test changes.

Reply:
That's a better idea. I'm used to the fabricator days when you couldn't do that!

Reply:
Done.

Reply:
> That's a better idea. I'm used to the fabricator days when you couldn't do that!

Splitting the test changes completely off the functional changes has a few advantages compared to adding them as separate commit to this PR:
1. Having a separate PR avoids cluttering the functional review with a large number of comments regarding the tests
2. Github's PR interface shows the changes from all commits in the `Files changed` view
3. As all commits in a PR get squashed together, the commit landing won't just show the diff on the tests caused by the patch, making it more difficult for people looking at the commit later (e.g. when doing post-commit review).

Reply:
The only problem is that you'll probably still end up with lots of comments or requests to change the tests once reviewers see the actual code changes and, for example, realise that a test case is missing for a particular change. Then you still end up changing the tests in the second patch anyway and at that point it's too late to change the first test-only patch because it has already landed.

I realise the official solution to this is to create branches in the main llvm repo in order to do stacked pull requests. That technically is a solution that probably works well for some cases, but given this is a very frequent thing to do it's also more costly and it seems that people aren't deleting their branches afterwards. `git branch -r -l | grep users | wc -l` returns a count of 638.

Not trying to say which way is right or wrong, or what works best for the individual developer, but just pointing out there are pros and cons to each approach.

Reply:
I agree with David's concern. Regarding your points fhahn, one can see the differences between commits in the "files changed" view and there is a link to the PR in the commit body so people can see the test diff during post-commit review.